lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -3011,3 +3011,71 @@ llvm::Error glow::optimizeFunction(Function *F, const Backend &B,
 
   return checkAllNodesSupported(*F, B);
 }
+
+bool glow::executeVerticalFCWeightsSplit(Function *F, unsigned numOfChunks,
+                                         unsigned minKToSplit) {
+  DCHECK(numOfChunks > 0) << "numOfChunks must be a positive number, given: "
+                          << numOfChunks;
+  DCHECK(minKToSplit > 0) << "minKToSplit must be a positive number, given: "
+                          << minKToSplit;
+
+  bool changed = false;
+  for (auto &node : F->getNodes()) {
```

Comment:
nit: Perhaps we should iterate over the Nodes backwards? Because we're iterating forward over all the Nodes in the Function, we will iterate over all of the newly added FCs from inside this again. This shouldn't cause infinite loop as is I believe because of the `K < minKToSplit` check. But things could change with this logic in the future. It's also slightly inefficient/unnecessary because we know we won't want to split those FCs again.